src/transport/encoding/raptorq/safe.rs

```diff
@@ -0,0 +1,118 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+/*
+ * The original implementation of this file can be found at:
+ * https://github.com/cberner/raptorq
+ *
+ * The original work is licensed under the Apache License, Version 2.0.
+ *
+ * Modifications made:
+ * - Add implementation for a safe ObjectTransmissionInformation
+ *   deserialization that checks parameter in order to prevent panic at
+ *   runtime
+ *
+ * The following is a notice for the original file:
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use std::convert::{TryFrom, TryInto};
+
+use raptorq::ObjectTransmissionInformation;
+
+use super::{encoder::MAX_MTU, TRANSMISSION_INFO_SIZE};
+
+// This should eventually become <https://doc.rust-lang.org/std/primitive.u64.html#method.div_ceil>
+// when it gets stabilized, and this function should be removed.
+// (1) the result is known to not overflow u32 from elsewhere;
+// (2) `denom` is known to not be `0` from elsewhere.
+// TODO this is definitely not always the case! Let's do something about it.
+fn int_div_ceil(num: u64, denom: u64) -> u32 {
+    if num % denom == 0 {
+        (num / denom) as u32
+    } else {
+        (num / denom + 1) as u32
+    }
```

Comment:
FYI, this can be simplified as

```suggestion
((num + denom - 1) / denom) as u32
```

Feel free to pick your favorite version

Reply:
If you change the code you don't have to keep the comments :)

Reply:
I would not change the code

Reply:
Its funny because the implementation that @fed-franz suggests is so well known. I understand not wanting to change a file you copied, but its pretty weird the raptor peeps don't know it.